### DISCUSSION TITLE
toggle hide table of contents with keypress.

### DISCUSSION THREAD

**Discussion by simonallfrey** [thegeeklab/hugo-geekdoc#408] [Ideas]
Sometimes I need a bit more space, so I've hacked in toggling hiding the table of contents by pressing 't'.

I'm lazy so I used jquery and dumped the code at the end of a site copy of site-footer.html
and in static/js/myscript.js

Perhaps people would appreciate this being implemented properly?

```
$ pwd
/tmp/geekdoc-demo

$ ls
archetypes  config.toml  content  data  layouts  nohup.out  public  resources  static  themes

$ tail -5 layouts/partials/site-footer.html
  </nav>
</footer>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
<script src="{{ "/js/myscripts.js" | urlize | relURL }}"></script>

$ cat static/js/myscripts.js
//hide the Table of Contents with the t key
$(document).bind('keydown',function(e) {
        if (e.key == 't'){
                //alert(e.key);
                $('aside').toggle();
        }
});
```

**Comment by xoxys** [comment]
Hi, thanks for the idea. What about just adding the TOC to the expand shortcode (there is a bug right now, but that can be fixed easily)?

![image](https://user-images.githubusercontent.com/3391958/169699347-8f3fcc85-6b98-49bc-bc1b-e8219db2c1df.png)

Example code:

```Markdown
{{< expand "Table of Content" >}}
{{< toc-tree >}}
{{< /expand >}}
```

**Comment by xoxys** [comment]
Bugs fixed, both toc shortcodes can be nested to the expand shortcode now.

```
{{< expand "Table of Content" >}}
{{< toc format=raw >}}
{{< /expand >}}
```

![image](https://user-images.githubusercontent.com/3391958/170517675-3d74d2c5-9a8b-4cbc-8070-82eceeedc790.png)
